### PR TITLE
ISPN-11480 Call EmbeddedCacheManager#stop in shutdown hook

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-11480
https://issues.redhat.com/browse/ISPN-11485

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/Server.java [hunk 1]
```diff
@@ -408,12 +409,15 @@ private void localShutdown(ExitStatus exitStatus) {
...
       protocolServers.values().parallelStream().forEach(ProtocolServer::stop);
-      cacheManagers.values().forEach(cm -> SecurityActions.stopCacheManager(cm));
+      cacheManagers.values().forEach(SecurityActions::stopCacheManager);
       this.status = ComponentStatus.TERMINATED;
       // Don't wait for the scheduler to finish
       if (scheduler != null) {
          scheduler.shutdown();
       }
+      // Shutdown Log4jk context manually as we set shutdownHook="disable" in order to disable the default Log4j
+      // shutdownHook being registered in order to allow logging during server shutdown
```
Comment (danberindei) [inline server/runtime/src/main/java/org/infinispan/server/Server.java]: A little too many "in order to" :)
I'd say something like
```
// Shutdown Log4jk context manually as we set shutdownHook="disable" 
// Log4j's shutdownHook may run concurrently with our shutdownHook,
// disabling logging before the server has finished stopping.
```

File: core/src/test/java/org/infinispan/statetransfer/StatefulSetRollingUpgradeTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      for (int id = 0; id < numNodes; id++)
+         createStatefulCacheManager(id);
+
+      waitForClusterToForm(CACHE_NAME);
+   }
+
+   public void testStateTransferRestart() {
+      for (int i = 0; i < NUM_ROLLING_UPGRADES; i++) {
+         for (int j = numNodes - 1; j > -1; j--) {
+            // Stop the cache manager in the same manner as via the ShutdownHook
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/statetransfer/StatefulSetRollingUpgradeTest.java]: The comment seems superfluous now.

Maybe we should call `waitForClusterToForm(CACHE_NAME)` before starting the joiner, too?

File: core/src/test/java/org/infinispan/statetransfer/StatefulSetRollingUpgradeTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * Start cluster (A,B) redeploy after upgrade. Rolling upgrades always occur in the order B,A and A does not restart
+ * until B has completed successfully.
+ *
+ * @author Ryan Emerson
+ * @since 11.0
+ */
+@CleanupAfterMethod
+@Test(groups = "functional", testName = "statetransfer.StateTransferRollingUpgradeTest")
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/statetransfer/StatefulSetRollingUpgradeTest.java]:
```suggestion
@Test(groups = "functional", testName = "statetransfer.StatefulSetRollingUpgradeTest")
```

**Comment by ryanemerson** [comment]
Updated

File: server/runtime/src/main/java/org/infinispan/server/Server.java [hunk 4]
```diff
@@ -409,12 +410,16 @@ private void localShutdown(ExitStatus exitStatus) {
...
+      cacheManagers.values().forEach(SecurityActions::stopCacheManager);
       this.status = ComponentStatus.TERMINATED;
       // Don't wait for the scheduler to finish
       if (scheduler != null) {
          scheduler.shutdown();
       }
+      // Shutdown Log4jk context manually as we set shutdownHook="disable"
+      // Log4j's shutdownHook may run concurrently with our shutdownHook,
+      // disabling logging before the server has finished stopping.
+      LogManager.shutdown();
```
Comment (wburns) [inline server/runtime/src/main/java/org/infinispan/server/Server.java]: Guess I never knew we required log4j in the server.

**Comment by wburns** [comment]
Integrated into master, thanks @ryanemerson !

**Pull request merged**